src/Account.sol

```diff
@@ -79,53 +100,51 @@ contract Account is IAccount, Auth, OpsReady {
     /// @notice track conditional orders by id
     mapping(uint256 id => ConditionalOrder order) internal conditionalOrders;
 
+    /// @notice value used for reentrancy protection
+    uint256 internal locked = 1;
+
     /*//////////////////////////////////////////////////////////////
                                MODIFIERS
     //////////////////////////////////////////////////////////////*/
 
     modifier isAccountExecutionEnabled() {
-        _isAccountExecutionEnabled();
+        if (!SETTINGS.accountExecutionEnabled()) {
+            revert AccountExecutionDisabled();
+        }
```

Comment:
**Gas Optimisation:"" It reduces the deployed bytecode to extract all logic in modifiers out of the modifier into a separate function. This is because otherwise the compiler wraps every function in the same code, but if you use helper functions it reuses the same function logic.

So the suggestion is this pattern:
```solidity
modifier isAccountExecutionEnabled() {
    _isAccountExecutionEnabled();
    _;
}

function _isAccountExecutionEnabled() {
    if (!SETTINGS.accountExecutionEnabled()) {
        revert AccountExecutionDisabled();
    }
}
```

Reply:
More expensive to have an internal function call* instead of leaving logic in the modifier

Reply:
Oh interesting, that is the opposite of what I expected!